File: session3/lecture63/dfs_practice_1.cpp
#include <bits/stdc++.h>
using namespace std;

const int n = 6;
vector<int> adj[n];
int visited[n];

void dfs(int u)
{
  visited[u] = 1;

  for (int nu : adj[u])
  {
    if (visited[u] == 0)
    {
      dfs(nu);
    }
  }
  return;
}

int main()
{
  adj[1].push_back(2);
  adj[1].push_back(3);
  adj[2].push_back(4);
  adj[2].push_back(5);
  adj[4].push_back(2);
  return 0;
}

File: session3/lecture63/dfs_practice_1.js
const n = 6;
const adj = Array.from({ length: n }, () => []);
const visited = new Array(n).fill(0);

function dfs(u) {
  visited[u] = 1;

  for (const nu of adj[u]) {
    if (visited[u] === 0) {
      dfs(nu);
    }
  }
  return;
}

adj[1].push(2);
adj[1].push(3);
adj[2].push(4);
adj[2].push(5);
adj[4].push(2);
